src/pudl/dbt_wrapper.py

```diff
@@ -75,14 +122,27 @@ def build_with_context(
     dbt = dbtRunner()
     dbt_dir = PUDL_ROOT_PATH / "dbt"
 
-    # 2025-06-06 TODO: use the --project-dir flag instead of chdiring all the time?
     with chdir(dbt_dir):
-        _ = dbt.invoke(["deps"])
-        _ = dbt.invoke(["seed"])
+        dbt.invoke(["deps"])
+        dbt.invoke(["seed"])
         build_output: dbtRunnerResult = dbt.invoke(["build"] + cli_args)
         build_results = cast(RunExecutionResult, build_output.result)
 
-    failed_nodes = get_failed_nodes(build_results)
-    failure_contexts = get_context_for_nodes(failed_nodes)
+    failed_nodes = __get_failed_nodes(build_results)
```

Comment:
Now that `BuildResult` is a proper class, I wonder if it would be a little cleaner to have all this logic about handling `build_results` contained in the `BuildResult` class. What I'm thinking is the `build_with_context` function would just handle the setup and running of the tests, then pass `build_output`/`build_results` to a `classmethod` on `BuildResult`, which figures out what to do with them.

Reply:
So basically move 131-145 into some sort of `BuildResult.from_run_execution_result()` function? Seems like it would make this function shorter, and make the logic reusable elsewhere, but also adds a layer of indirection here... sort of 6 of one, half dozen of the other in my book. I'm going to keep this as is since I worry that if I change it something unexpectedly subtle will break and I might as well just merge this guy.